src/ucs/datastruct/bitmap.h

```diff
@@ -297,19 +299,26 @@ typedef uint64_t ucs_bitmap_word_t;
     ({ \
         size_t _word_index = 0, _popcount = 0; \
         _UCS_BITMAP_FOR_EACH_WORD(_bitmap, _word_index) { \
-            if (_bit_index >= ((_word_index) + 1) * UCS_BITMAP_BITS_IN_WORD) { \
+            if ((_bit_index) >= ((_word_index) + 1) * UCS_BITMAP_BITS_IN_WORD) { \
                 _popcount += ucs_popcount( \
                     _UCS_BITMAP_WORD(_bitmap, _word_index)); \
             } else { \
                 _popcount += ucs_popcount( \
                     _UCS_BITMAP_WORD(_bitmap, _word_index) & \
-                    (UCS_MASK(_bit_index % UCS_BITMAP_BITS_IN_WORD))); \
+                    (UCS_MASK((_bit_index) % UCS_BITMAP_BITS_IN_WORD))); \
             } \
         } \
         _popcount; \
     })
 
 
+/**
+ *  Mask a single word in a bitmap using a given mask
```

Comment:
Return a word-mask for the word at '_word_index' for all the bits up to (and not including) '_mask_index'.